src/ResourceManager/Sql/Commands.Sql/Elastic Pools/Cmdlet/SetAzureSqlElasticPool.cs

```diff
@@ -45,41 +48,85 @@ public class SetAzureSqlElasticPool : AzureSqlElasticPoolCmdletBase
         /// </summary>
-        [Parameter(Mandatory = false,
+        [Parameter(ParameterSetName = DtuPoolParameterSet, Mandatory = false,
             HelpMessage = "The total shared DTU for the Sql Azure Elastic Pool.")]
         [ValidateNotNullOrEmpty]
         public int Dtu { get; set; }
 
         /// <summary>
         /// Gets or sets the storage limit for the Sql Azure Database Elastic Pool in MB.
         /// </summary>
-        [Parameter(Mandatory = false,
+        [Parameter(ParameterSetName = DtuPoolParameterSet, Mandatory = false,
             HelpMessage = "The storage limit for the Sql Azure Elastic Pool in MB.")]
+        [Parameter(ParameterSetName = VcorePoolParameterSet, Mandatory = false, HelpMessage = "The storage limit for the Sql Azure Elastic Pool in MB.")]
         [ValidateNotNullOrEmpty]
         public int StorageMB { get; set; }
 
         /// <summary>
         /// Gets or sets the minimum DTU all Sql Azure Databases are guaranteed.
         /// </summary>
-        [Parameter(Mandatory = false,
+        [Parameter(ParameterSetName = DtuPoolParameterSet, Mandatory = false,
             HelpMessage = "The minimum DTU all Sql Azure Databases are guaranteed.")]
         [ValidateNotNullOrEmpty]
         public int DatabaseDtuMin { get; set; }
 
         /// <summary>
         /// Gets or sets the maximum DTU any one Sql Azure Database can consume.
         /// </summary>
-        [Parameter(Mandatory = false,
+        [Parameter(ParameterSetName = DtuPoolParameterSet, Mandatory = false,
             HelpMessage = "The maximum DTU any one Sql Azure Database can consume.")]
         [ValidateNotNullOrEmpty]
         public int DatabaseDtuMax { get; set; }
 
+        /// <summary>
+        /// Gets or sets the total shared VCore number for the Sql Azure Elastic Pool.
+        /// </summary>
+        [Parameter(ParameterSetName = VcorePoolParameterSet, Mandatory = false,
+            HelpMessage = "The total shared number of Vcore for the Sql Azure Elastic Pool.")]
+        [ValidateNotNullOrEmpty]
+        public int VCore { get; set; }
+
+        /// <summary>
+        /// Gets or sets the compute generation for the Sql Azure Elastic Pool
+        ///   (Available ComputeGeneration in the format of: Gen4, Gen5).
+        /// </summary>
+        [Parameter(ParameterSetName = VcorePoolParameterSet, Mandatory = false,
+            HelpMessage = "The compute generation for the Sql Azure Elastic Pool. e.g. 'Gen4', 'Gen5'.")]
+        [ValidateNotNullOrEmpty]
+        [Alias("Family")]
+        [PSArgumentCompleter("Gen4", "Gen5")]
+        public string ComputeGeneration { get; set; }
+
+        /// <summary>
+        /// Gets or sets the minimum vcore any database can consume in the pool.
+        /// </summary>
+        [Parameter(ParameterSetName = VcorePoolParameterSet, Mandatory = false,
+            HelpMessage = "The minimum VCore number any SqlAzure Database can consume in the pool.")]
```

Comment:
Probably don't need to repeat "Sql Azure" in all the help messages.

Reply:
Additionally in some places you have 'Sql Azure' in others you have 'SqlAzure'. It should be 'Azure SQL'